### PR TITLE
feat: update squeeze page record

### PR DESCRIPTION
**Pull request by coderoyalty**
## Description: What this PR is.

This PR is about updating a squeeze page record.

## How it can be tested.

Make a `PUT` request to the endpoint `/api/v1/squeeze` with a request body matching this:

```json
{
  "email": "user@example.com",
  "first_name": "John",
  "last_name": "Doe",
  "phone": "08098761234",
  "location": "Lagos, Nigeria",
  "job_title": "Software Engineer",
  "company": "X-Corp",
  "interests": ["Web Development", "Cloud Computing"],
  "referral_source": "LinkedIn"
}
```

Response:

```json
{
  "message": "Your record has been successfully updated. You cannot update it again.",
  "status_code" 200
}
```

## Checklist of What I did

- [x] Provide RESTful API endpoint to handle `PUT` request
- [x] Validate and verify incoming request body
- [x] Mitigate user from updating their email address
- [x] Correctly update the squeeze page record (return the appropriate response for failed operations)
- [x] Handle errors and return appropriate status codes.
- [x] Make sure a squeeze record is not updated more than once.

## Reference Issue
#430 : [FEAT] Implement API endpoint for updating squeeze page record

### CODE REVIEW

**Review by Heba-WebDev — APPROVED** [review]
Looks good to me.

**Comment by buka4rill** [comment]
Fix conflicts

File: src/modules/squeeze/squeeze.controller.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import { UpdateSqueezeDto } from './dto/update-squeeze.dto';
+import { ApiResponse, ApiOperation, ApiTags } from '@nestjs/swagger';
+import { skipAuth } from '../../helpers/skipAuth';
+
+@ApiTags('Squeeze')
+@Controller('squeeze')
+export class SqueezeController {
+  constructor(private readonly squeezeService: SqueezeService) {}
+
+  @skipAuth()
```
Comment (Homoakin619) [inline src/modules/squeeze/squeeze.controller.ts:12]: is this route a protected route ? if yes remove the skipAuth() decorator from here.
The function of the decorator is to override authentication

Comment (coderoyalty) [reply src/modules/squeeze/squeeze.controller.ts:12]: It's not a protected route.

File: src/modules/squeeze/squeeze.service.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+      if (!squeeze) {
+        throw new NotFoundException({
+          message: 'No squeeze page record exists for the provided request body',
+          status_code: 404,
+        });
+      }
+
+      if (squeeze.updated_at > squeeze.created_at) {
+        // the squeeze was previously updated
```
Comment (Homoakin619) [inline src/modules/squeeze/squeeze.service.ts:26]: remove comment

File: src/modules/squeeze/squeeze.controller.ts (hunk 1 above)
Comment (Homoakin619) [reply src/modules/squeeze/squeeze.controller.ts:12]: alright

**Review by Homoakin619 — DISMISSED** [review]

**Review by davidolanrewaju — APPROVED** [review]

**Review by Heba-WebDev — DISMISSED** [review]

File: src/modules/squeeze/squeeze.controller.ts (hunk 1 above)
Comment (Homoakin619) [inline src/modules/squeeze/squeeze.controller.ts:12]: Is this route protected? if yes remove this decorator

File: src/modules/squeeze/squeeze.controller.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  })
+  @ApiResponse({
+    status: 404,
+    description: 'No squeeze page record exists for the provided request body',
+  })
+  @ApiResponse({
+    description: 'The squeeze page record can only be updated once.',
+    status: 403,
+  })
+  async update(@Body() updateDto: UpdateSqueezeDto) {
```
Comment (Homoakin619) [inline src/modules/squeeze/squeeze.controller.ts:33]: make this method name more descriptive such as updateSqueeze

File: src/modules/squeeze/squeeze.controller.ts (hunk 1 above)
Comment (coderoyalty) [reply src/modules/squeeze/squeeze.controller.ts:12]: it's not protected.

File: src/modules/squeeze/squeeze.controller.ts (hunk 3 above)
Comment (coderoyalty) [reply src/modules/squeeze/squeeze.controller.ts]: done!

**Review by Homoakin619 — APPROVED** [review]

**Review by Heba-WebDev — DISMISSED** [review]

**Comment by coderoyalty** [comment]
The changes this PR #518 is trying to make, already exist within this PR (#438), and will close issue #438

**Review by Heba-WebDev — APPROVED** [review]

**Review by alexindevs — DISMISSED** [review]

**Comment by alexindevs** [comment]
This branch has merge conflicts ❌

**Review by Cobidex — COMMENTED** [review]
Fix merge conflicts

**Comment by coderoyalty** [comment]
@alexindevs @Cobidex I've resolved the merge conflict.

**Review by Heba-WebDev — DISMISSED** [review]

**Review by Heba-WebDev — DISMISSED** [review]

**Review by Heba-WebDev — DISMISSED** [review]

**Review by Homoakin619 — APPROVED** [review]

**Review by davidolanrewaju — APPROVED** [review]

**Review by TheCodeGhinux — APPROVED** [review]

**Review by Heba-WebDev — APPROVED** [review]

**Review by Cobidex — APPROVED** [review]

**Pull request merged**